src/lib.rs

```diff
@@ -490,13 +458,35 @@ pub struct Section {
     final_size: Label,
 }
 
+/// Indicates that there were one or more undefined labels.
+///
+/// The locations at which these labels were created are in the Debug implementation of this
+/// struct.
+pub struct UndefinedLabelsError {
+    locations: Vec<StaticLocation>,
+}
+
+impl std::fmt::Debug for UndefinedLabelsError {
```

Comment:
If we're going to use this as the error type in a `Result`, we should probably go ahead and `impl std::error::Error` for it: https://doc.rust-lang.org/std/error/trait.Error.html
It's not a big deal and 3/4 of the methods can be stubbed out to return `None`. It does require you to also `impl Display`, but it's fine to make `Debug` and `Display` have the same output and have one impl call the other.

Reply:
Sounds good, yep.

Actually, *all* of them can be. (And we *can't* implement `description` other than rendering ourselves to a String when we're created, which is pretty terrible. Hence why it's deprecated now.)